optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/manager/DefaultSolverManager.java

```diff
@@ -0,0 +1,242 @@
+        Objects.requireNonNull(solverConfigResource);
+        Objects.requireNonNull(classLoader);
+        return new DefaultSolverManager<>(solverConfigResource, classLoader, null, null);
+    }
+
+    public static <Solution_> SolverManager<Solution_> createFromXmlResource(String solverConfigResource, ThreadFactory threadFactory) {
+        Objects.requireNonNull(solverConfigResource);
+        Objects.requireNonNull(threadFactory);
+        return new DefaultSolverManager<>(solverConfigResource, null, null, threadFactory);
+    }
+
+    public static <Solution_> SolverManager<Solution_> createFromXmlResource(
+            String solverConfigResource,
+            ClassLoader classLoader,
+            ThreadFactory threadFactory) {
+        Objects.requireNonNull(solverConfigResource);
+        Objects.requireNonNull(classLoader);
+        Objects.requireNonNull(threadFactory);
+        return new DefaultSolverManager<>(solverConfigResource, classLoader, null, threadFactory);
+    }
+
+    public static <Solution_> SolverManager<Solution_> createFromSolverFactory(SolverFactory<Solution_> solverFactory) {
+        return new DefaultSolverManager<>(null, null, solverFactory, null);
+    }
+
+    public static <Solution_> SolverManager<Solution_> createFromSolverFactory(
+            SolverFactory<Solution_> solverFactory,
+            ThreadFactory threadFactory) {
+        Objects.requireNonNull(solverFactory);
+        Objects.requireNonNull(threadFactory);
+        return new DefaultSolverManager<>(null, null, solverFactory, threadFactory);
+    }
+
+    private DefaultSolverManager(String solverConfigResource, ClassLoader classLoader, SolverFactory<Solution_> solverFactory, ThreadFactory threadFactory) {
+        if (solverFactory != null) {
+            this.solverFactory = solverFactory;
+        } else if (classLoader != null) {
+            this.solverFactory = SolverFactory.createFromXmlResource(solverConfigResource, classLoader);
+        } else {
+            this.solverFactory = SolverFactory.createFromXmlResource(solverConfigResource);
+        }
+
+        int numAvailableProcessors = Runtime.getRuntime().availableProcessors();
+        int activeThreadCount = Math.max(numAvailableProcessors - 2, 1);
+        logger.info("Number of available processors: {}. Active thread count: {}.", numAvailableProcessors, activeThreadCount);
+
+        if (threadFactory != null) {
+            this.solverFactory.getSolverConfig().setThreadFactoryClass(threadFactory.getClass());
+            solverExecutorService = Executors.newFixedThreadPool(activeThreadCount, threadFactory); // #threads < #cpus implies a FIFO scheduling policy
+            eventHandlerExecutorService = Executors.newSingleThreadExecutor(threadFactory);
+        } else {
+            solverExecutorService = Executors.newFixedThreadPool(numAvailableProcessors - 1);
```

Comment:
Is the `numAvailableProcessors - 1` intended? I don't see a reason why the number of threads in the pool should be different depending on whether you supply a thread factory or not.

Reply:
For example, if `#cpus == 1` then we'll have a thread pool of a zero size. So I'd really expect `activeThreadCount` here.

Reply:
Nice catch, you're right it should be `activeThreadCount`, I updated it in one place and missed the other. Thanks!